{"cells":[{"cell_type":"code","execution_count":5,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"-pgqY1dwNHXR","executionInfo":{"status":"ok","timestamp":1642225726743,"user_tz":240,"elapsed":30058,"user":{"displayName":"Ivan Jose Duque Luna","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiJhoMRg6vZLPIXpslz4bHdxaLre7D1LZX_BfQT=s64","userId":"01917523343114875724"}},"outputId":"d22bcdbe-953a-4e03-b4eb-c242de4cf935"},"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting face_recognition\n","  Downloading face_recognition-1.3.0-py2.py3-none-any.whl (15 kB)\n","Requirement already satisfied: Click>=6.0 in /usr/local/lib/python3.7/dist-packages (from face_recognition) (7.1.2)\n","Collecting face-recognition-models>=0.3.0\n","  Downloading face_recognition_models-0.3.0.tar.gz (100.1 MB)\n","\u001b[K     |████████████████████████████████| 100.1 MB 30 kB/s \n","\u001b[?25hRequirement already satisfied: Pillow in /usr/local/lib/python3.7/dist-packages (from face_recognition) (7.1.2)\n","Requirement already satisfied: dlib>=19.7 in /usr/local/lib/python3.7/dist-packages (from face_recognition) (19.18.0)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.7/dist-packages (from face_recognition) (1.19.5)\n","Building wheels for collected packages: face-recognition-models\n","  Building wheel for face-recognition-models (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Created wheel for face-recognition-models: filename=face_recognition_models-0.3.0-py2.py3-none-any.whl size=100566186 sha256=397fa7b223a546e8b606a7dbe0d4c3e92075bed3713afcb888b3d2cb14666499\n","  Stored in directory: /root/.cache/pip/wheels/d6/81/3c/884bcd5e1c120ff548d57c2ecc9ebf3281c9a6f7c0e7e7947a\n","Successfully built face-recognition-models\n","Installing collected packages: face-recognition-models, face-recognition\n","Successfully installed face-recognition-1.3.0 face-recognition-models-0.3.0\n","E: Unable to locate package face_recognition\n"]}],"source":["!pip install face_recognition\n","!apt-get -qq install -y face_recognition"]},{"cell_type":"code","source":["import face_recognition\n","import numpy as np"],"metadata":{"id":"yMtLD4UbRye-","executionInfo":{"status":"ok","timestamp":1642225731186,"user_tz":240,"elapsed":3193,"user":{"displayName":"Ivan Jose Duque Luna","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiJhoMRg6vZLPIXpslz4bHdxaLre7D1LZX_BfQT=s64","userId":"01917523343114875724"}}},"execution_count":6,"outputs":[]},{"cell_type":"code","execution_count":7,"metadata":{"id":"PsWvstM-OPpB","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1642225735692,"user_tz":240,"elapsed":3503,"user":{"displayName":"Ivan Jose Duque Luna","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiJhoMRg6vZLPIXpslz4bHdxaLre7D1LZX_BfQT=s64","userId":"01917523343114875724"}},"outputId":"8be5a659-0b73-47a0-b19d-9d66295333dd"},"outputs":[{"output_type":"stream","name":"stdout","text":["(325, 325, 3)\n","[(63, 526, 384, 205)]\n","(128,)\n","-----------------------\n","A face is located at pixel location Top: 63, Left: 205, Bottom: 384, Right: 526\n","[True]\n"]}],"source":["\n","image = face_recognition.load_image_file(\"rostrocortado25.jpg\")\n","image2 = face_recognition.load_image_file(\"rostrocompleto20.jpg\")\n","\n","print(image.shape)\n","\n","\n","face_locations = face_recognition.face_locations(image2)\n","print(face_locations)\n","\n","face_encoding = face_recognition.face_encodings(image)[0]\n","face_encoding2 = face_recognition.face_encodings(image2)[0]\n","\n","print(face_encoding.shape)\n","\n","print(\"-----------------------\")\n","#print(\"I found {} face(s) in this photograph.\".format(len(face_locations)))\n","\n","for face_location in face_locations:\n","\n","    # Print the location of each face in this image\n","    top, right, bottom, left = face_location\n","    print(\"A face is located at pixel location Top: {}, Left: {}, Bottom: {}, Right: {}\".format(top, left, bottom, right))\n","\n","    # You can access the actual face itself like this:\n","    # face_image = image[top:bottom, left:right]\n","    # pil_image = Image.fromarray(face_image)\n","    # pil_image.show()\n","\n","results = face_recognition.compare_faces([face_encoding], face_encoding2)\n","\n","print(results)"]}],"metadata":{"colab":{"name":"probando face recognition.ipynb","provenance":[],"authorship_tag":"ABX9TyO+LiZkyMZBH0uRg9/4p5od"},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"},"accelerator":"GPU"},"nbformat":4,"nbformat_minor":0}